src/core/agent/services/ipexCommunicationService.test.ts

```diff
@@ -892,41 +924,111 @@ describe("Ipex communication service of agent", () => {
     });
     expect(
-      Agent.agent.signifyNotifications.addPendingOperationToQueue
+      signifyNotificationService.addPendingOperationToQueue
     ).toBeCalledTimes(1);
-    expect(deleteNotificationMock).toBeCalledWith(id);
   });
 
-  test("cannot accept ACDC from multisig exn if the notification is missing in the DB", async () => {
-    Agent.agent.getKeriaOnlineStatus = jest.fn().mockReturnValueOnce(true);
-    const id = "not-found-id";
-    notificationStorage.findById.mockResolvedValueOnce(null);
+  test("can accept ACDC from multisig exn when existing credential", async () => {
+    Agent.agent.getKeriaOnlineStatus = jest.fn().mockReturnValue(true);
+    const id = "uuid";
+    getExchangeMock
+      .mockReturnValueOnce({
+        exn: {
+          v: "KERI10JSON00032d_",
+          t: "exn",
+          d: "ELW97_QXT2MWtsmWLCSR8RBzH-dcyF2gTJvt72I0wEFO",
+          i: "ECS7jn05fIP_JK1Ub4E6hPviRKEdC55QhxZToxDIHo_E",
+          rp: "EJ84hiNC0ts71HARE1ZkcnYAFJP0s-RiLNyzupnk7edn",
+          p: "",
+          dt: "2024-08-28T06:39:55.501000+00:00",
+          r: "/multisig/exn",
+          q: {},
+          a: {
+            i: "EJ84hiNC0ts71HARE1ZkcnYAFJP0s-RiLNyzupnk7edn",
+            gid: "EC1cyV3zLnGs4B9AYgoGNjXESyQZrBWygz3jLlRD30bR",
+          },
+          e: {
+            exn: {
+              v: "KERI10JSON000178_",
+              t: "exn",
+              d: "EKa94ERqArLOvNf9AmItMJtsoGKZPVb3e_pEo_1D37qt",
+              i: "EC1cyV3zLnGs4B9AYgoGNjXESyQZrBWygz3jLlRD30bR",
+              rp: "EJ84hiNC0ts71HARE1ZkcnYAFJP0s-RiLNyzupnk7edn",
+              p: "EAe_JgQ636ic-k34aUQMjDFPp6Zd350gEsQA6HePBU5W",
+              dt: "2024-08-28T06:39:51.416000+00:00",
+              r: "/ipex/admit",
+              q: {},
+              a: {
+                i: "EJ84hiNC0ts71HARE1ZkcnYAFJP0s-RiLNyzupnk7edn",
+                m: "",
+              },
+              e: {},
+            },
+            d: "EE8_Xc0ZUh_sUJLtmBpVSEr-RFS2mRUIpFyL-pmvtPvx",
+          },
+        },
+        pathed: {},
+      })
+      .mockReturnValueOnce(grantIpexMessageMock);
+    notificationStorage.findAllByQuery = jest.fn().mockResolvedValue([
+      {
+        type: "NotificationRecord",
+        id: id,
+        createdAt: new Date(),
+        a: {
+          r: "/exn/ipex/grant",
+          d: "EIDUavcmyHBseNZAdAHR3SF8QMfX1kSJ3Ct0OqS0-HCW",
+        },
+        route: "/exn/ipex/grant",
+        read: true,
+        linkedGroupRequests: {
+          "EDm8iNyZ9I3P93jb0lFtL6DJD-4Mtd2zw1ADFOoEQAqw": false,
+        },
+        connectionId: "EEFjBBDcUM2IWpNF7OclCme_bE76yKE3hzULLzTOFE8E",
+        updatedAt: new Date(),
+      },
+    ]);
 
-    await expect(
-      ipexCommunicationService.acceptAcdcFromMultisigExn(id)
-    ).rejects.toThrowError(
-      `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
-    );
+    await ipexCommunicationService.acceptAcdc(id);
+
+    expect(operationPendingStorage.save).toBeCalledWith({
```

Comment:
need to expect that we DON'T save the pending record because it already exists

Reply:
This is still missing - so we check that `saveCredentialMetadataRecord` is not called because it already exists, and it is called when it doesn't exist

Reply:
pushed, please check it again